client/src/screens/SignIn/SignIn.jsx

```diff
@@ -1,35 +1,105 @@
-import { useState } from 'react';
+import { useEffect, useState } from 'react';
 import { useNavigate } from 'react-router-dom';
-import { Header } from '../../components/Header/Header.jsx';
-import { Form } from '../../components/Form/Form';
 // assets
 import { useDispatch } from 'react-redux';
 import { loginUser } from '../../services/redux/slices/uiActions.js';
-import { signInForm } from '../../services/formData.js';
+import '../SignUp/SignUp.scss';
+import store from '../../services/redux/store.js'
+import { GenericModal } from '../../components/Modal/GenericModal';
+import { handleChange } from '../../services/utils/formHandlers';
+import { SingleActionButton } from '../../components/Button/SingleActionButton';
+import { checkEmailAuth, signOut } from '../../services/api/users';
 
 export const SignIn = () => {
   const navigate = useNavigate();
   const dispatch = useDispatch();
+  const [noAccountError, setNoAccountError] = useState(false);
+  const [showModal, setShowModal] = useState(false);
   const [loginInfo, setLoginInfo] = useState({
     email: 'test@test.com',
     password: 'test',
   });
 
+  useEffect( async () => {
+    signOut();
+  },[])
+
   const handleSignIn = async (event) => {
     event.preventDefault();
-    dispatch(loginUser(loginInfo));
-    navigate('/');
+    if (!noAccountError) {
+      await dispatch(loginUser(loginInfo));
+      //check for user
+      const state = store.getState();
```

Comment:
while this taught me that we can access the redux store using a different approach, i'd like to stick to using `useSelector` to access the store for 2 reasons:
- it might disrupt the redux lifecycle 
- this would import the entire redux store to this component, where the useSelector most likely just extracts the specific state we need